scripts/vmpublish.rb

```diff
@@ -0,0 +1,80 @@
+require 'logger'
+require 'pathname'
+require 'trollop'
+require 'json'
+
+require_relative 'config'
+
+
+def find_imgfac_image(image)
+  f = File.new(image)
+  Dir.chdir(STORAGE_DIR) do
+    Dir.glob('*.body') do |filename|
+      return filename.split('.').first if File.new(filename).stat.ino == f.stat.ino
+    end
+  end
+end
+
+
+def upload_imgfac_image(uuid, target, reference, version)
+  if target == 'vagrant'
+    creds_file = CREDS_DIR.join('atlas.json')
+    params = "atlas @ignored #{creds_file} --id #{uuid}"
+    params += " --parameter atlas_box_name #{reference} --parameter atlas_box_version #{version}"
+  elsif target == 'gce'
+    creds_file = CREDS_DIR.join('gce.json')
+    params = "gce @manageiq #{creds_file} --id #{uuid}"
+    params += " --parameter gce_object_name manageiq-#{reference}-#{version}.tar.gz"
+    params += " --parameter gce_image_name manageiq-#{reference}-#{version} --parameter gce_image_family centos-7"
+  end
+
+  if !File.exist?(creds_file)
+    $log.info "credentials file #{creds_file} does not exist, skipping #{target} publish"
+    return
+  end
+
+  Dir.chdir(IMGFAC_DIR) do
+    # Show progress on STDOUT
+    command = "./imagefactory provider_image #{params}"
+    $log.info "running command: #{command}"
+    if !system(command)
+      $log.error "imagefactory exited with status #{$?.exitstatus}"
+      exit 1
+    end
+  end
+end
+
+
+$log = Logger.new(STDOUT)
+
+dir_desc  = "Directory builds were copied to"
+type_desc = "Build type (stable or latest)"
+
+options = Trollop.options(ARGV) do
+  banner "Usage: vmpublish.rb [options]"
+  opt :copy_dir,  dir_desc,   :type => :string,   :short => 'd', :default => "master"
+  opt :type,      type_desc,  :type => :string,   :short => 't', :default => "stable"
```

Comment:
Same question as above, does publish belong as a step in the build process, if so, we should move the ARGV options there.